dvc/repo/update.py

```diff
@@ -1,11 +1,17 @@
+from dvc.exceptions import UpdateWithRevNotPossibleError
+
 from . import locked
 
 
 @locked
-def update(self, target):
+def update(self, target, rev=None):
     from dvc.stage import Stage
 
     stage = Stage.load(self, target)
-    stage.update()
+
+    if not stage.is_repo_import and rev:
+        raise UpdateWithRevNotPossibleError()
+
```

Comment:
We shouldn't check it here, but rather in `dvc/stage.py`, similar to the check that we have there already.

Reply:
yeah, my bad. it is better to do this inside `Stage.update()` so can be reused!

Reply:
done by 9d68c8aa